### PR TITLE
avoid copying ActorTableData when NodeMananger updates an actor to GCS

### PR DESCRIPTION
**Pull request by micafan**
## What do these changes do?

1. Avoid copying ActorTableData when NodeMananger updates/registers an actor to GCS
2. Fix comments

## Linter

- [x] I've run `scripts/format.sh` to lint the changes in this PR.

### CODE REVIEW

**Review by zhijunfu — APPROVED** [review]

File: src/ray/raylet/node_manager.cc [hunk 1]
```diff
@@ -1806,45 +1806,46 @@ void NodeManager::FinishAssignedTask(Worker &worker) {
...
   auto actor_id = task_spec.ActorCreationId();
   auto actor_entry = actor_registry_.find(actor_id);
-  ActorTableData new_actor_data;
+  std::shared_ptr<ActorTableData> actor_info_ptr;
   // TODO(swang): If this is an actor that was reconstructed, and previous
   // actor notifications were delayed, then this node may not have an entry for
   // the actor in actor_regisry_. Then, the fields for the number of
   // reconstructions will be wrong.
   if (actor_entry == actor_registry_.end()) {
+    actor_info_ptr.reset(new ActorTableData());
```
Comment (jovany-wang) [inline src/ray/raylet/node_manager.cc]: make_shared?

**Review by jovany-wang — APPROVED** [review]
LGTM

File: src/ray/raylet/node_manager.cc (hunk 1 above)
Comment (micafan) [reply src/ray/raylet/node_manager.cc]: One more copy.

Comment (jovany-wang) [reply src/ray/raylet/node_manager.cc]: No matter to copy a smart pointer here I think.

**Review by raulchen — APPROVED** [review]
LGTM. Can you resolve the conflicts? and then we can merge it.

File: src/ray/raylet/node_manager.cc (hunk 1 above)
Comment (jovany-wang) [reply src/ray/raylet/node_manager.cc]: https://stackoverflow.com/questions/36477022/boost-initializing-shared-pointer-reset-vs-make-shared/36477600

In general, make_shared will be more efficient I think.

**Pull request merged**